.github/workflows/winget.yml

```diff
@@ -0,0 +1,55 @@
+name: Publish on Windows Package Manager
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+    inputs:
+      tag_name:
+        description: Release tag (version)
+        required: true
+
+jobs:
+  publish-winget:
+    name: Publish
+    runs-on: windows-latest
+    if: github.event_name == 'workflow_dispatch' || github.event_name == 'release' && !github.event.release.prerelease
+    steps:
+      - name: Configure parameters
+        id: params
+        run: |
+          if ("${{ github.event_name }}" -eq "release") {
+            # Sometimes assets aren't immediately available, so we wait a bit for the first run
+            ${{ github.run_attempt == 1 && 'Start-Sleep -Seconds 300' || '' }}
+            
+            echo "assets_url=${{ github.event.release.assets_url }}" >> $env:GITHUB_OUTPUT
+            echo "version=${{ github.event.release.tag_name }}" >> $env:GITHUB_OUTPUT
+          } else {
+            $releaseUrl = "$env:GITHUB_API_URL/repos/clangd/clangd/releases/tags/${{ github.event.inputs.tag_name }}"
+            $assetsUrl = curl -sL $releaseUrl | ConvertFrom-Json | Select-Object -ExpandProperty assets_url
```

Comment:
If I'm understanding correctly, querying `assets_url` returns the same contents that are already present in the `assets` field of the `releaseUrl` response. Can we take the information from the `releaseUrl` response directly, rather than making a second request?

Reply:
Right, but `assets` aren't always available there, based on my experience.